copper-server-impl/src/main/java/com/pogeyan/cmis/impl/services/CmisVersioningServices.java

```diff
@@ -206,78 +210,95 @@ public static String checkOut(String repositoryId, Holder<String> objectId, Exte
 				throw new CmisObjectNotFoundException(
 						TracingWriter.log(String.format(ErrorMessages.UNKNOWN_OBJECT, objectId), span));
 			}
+			ITypePermissionService typePermissionFlow = TypeServiceFactory
+					.createTypePermissionFlowService(repositoryId);
+			boolean permission = CmisTypeServices.checkCrudPermission(typePermissionFlow, repositoryId, userObject,
```

Comment:
In all CREATE or UPDATE or DELETE checks, lets also check for READ. Once we change the last param as array then it is easy.